### PR TITLE
feat: [PIE-2578]: replaced error response for crud operations

### PR DESCRIPTION
**Pull request by goyalsonali**
Summary:

Changed error response of input and overlayinputsetform
Replaced them with the response receiving from backend

Jira Links:

https://harness.atlassian.net/browse/PIE-2578

 Screenshots:

Screenshot 2022-02-11 at 6 30 58 PM

Screenshot 2022-02-09 at 7 34 13 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/70-pipeline/components/InputSetForm/InputSetForm.tsx [hunk 1]
```diff
@@ -421,24 +421,18 @@ export const InputSetForm: React.FC<InputSetFormProps> = (props): JSX.Element =>
...
-            // This is done because when git sync is enabled, errors are displayed in a modal
-          } else if (!isGitSyncEnabled) {
-            showError(getString('inputSets.inputSetSavedError'), undefined, 'pipeline.create.inputset')
-          }
-        } else {
-          if (!isGitSyncEnabled) {
-            showSuccess(getString('inputSets.inputSetSaved'))
-            history.goBack()
-          }
+        if (!isGitSyncEnabled) {
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/InputSetForm/InputSetForm.tsx:424]: if (response && !isGitSyncEnabled))

File: src/modules/70-pipeline/components/InputSetForm/InputSetForm.tsx [hunk 2]
```diff
@@ -421,24 +421,18 @@ export const InputSetForm: React.FC<InputSetFormProps> = (props): JSX.Element =>
...
+          showSuccess(getString('inputSets.inputSetSaved'))
+          history.goBack()
         }
       }
     } catch (e) {
+      const errors = getFormattedErrors(e.data.metadata?.uuidToErrorResponseMap)
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/InputSetForm/InputSetForm.tsx:430]: e?.data?.meta?.

File: src/modules/70-pipeline/components/InputSetForm/InputSetForm.tsx [hunk 3]
```diff
@@ -421,24 +421,18 @@ export const InputSetForm: React.FC<InputSetFormProps> = (props): JSX.Element =>
...
+      if (Object.keys(errors).length) {
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/InputSetForm/InputSetForm.tsx:431]: use lodash's isEmpty

if (!isEmpty(errors)))

File: src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx [hunk 4]
```diff
@@ -404,29 +397,24 @@ export const OverlayInputSetForm: React.FC<OverlayInputSetFormProps> = ({
...
-              }
-            }
-          }
         }
       }
       if (!isGitSyncEnabled) {
         closeForm()
       }
     } catch (e: any) {
+      const getError = e.data?.metadata?.invalidReferences
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx:406]: naming should be in sync

File: src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx [hunk 5]
```diff
@@ -404,29 +397,24 @@ export const OverlayInputSetForm: React.FC<OverlayInputSetFormProps> = ({
...
+      const errors = Object.keys(getError ?? {}).filter((key: string) => {
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx:407]: isEmpty()

File: src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx [hunk 6]
```diff
@@ -404,29 +397,24 @@ export const OverlayInputSetForm: React.FC<OverlayInputSetFormProps> = ({
...
+        return {
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx:408]: if (getError?.[key]) {
   return {
      fieldId: ..,
      error: ...
   }
}

File: src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx [hunk 7]
```diff
@@ -404,29 +397,24 @@ export const OverlayInputSetForm: React.FC<OverlayInputSetFormProps> = ({
...
+          fieldId: key,
+          error: getError?.[key]
+        }
+      })
+      if (Object.keys(errors).length) {
```
Comment (chiragp1212) [inline src/modules/70-pipeline/components/OverlayInputSetForm/OverlayInputSetForm.tsx:413]: isEmpty

**Comment by goyalsonali** [comment]
retrigger jest

**Comment by goyalsonali** [comment]
retrigger coverage

**Comment by goyalsonali** [comment]
trigger cypress

**Comment by goyalsonali** [comment]
retrigger coverage

**Review by kapilkumawat86 — APPROVED** [review]

**Review by chiragp1212 — APPROVED** [review]

**Pull request merged**